subsys/testsuite/ztest/src/ztest.c

```diff
@@ -427,6 +429,37 @@ void end_report(void)
 K_APPMEM_PARTITION_DEFINE(ztest_mem_partition);
 #endif
 
+struct unit_test_node *z_ztest_unit_test_head;
+
+int ztest_run_registered_test_suites(const void *state)
+{
+	struct ztest_suite_node *ptr;
+	int count = 0;
+
+	for (ptr = (struct ztest_suite_node *)__ztest_suite_start; ptr < __ztest_suite_end; ++ptr) {
+		bool should_run = true;
+
+		if (ptr->pragma != NULL) {
+			should_run = ptr->pragma(state);
+		} else if (ptr->metadata.run_count > 0) {
+			/* If pragma is NULL and we already ran this test, skip it. */
+			should_run = false;
+		}
+
+		if (should_run) {
+			int fail = z_ztest_run_test_suite(ptr->name, ptr->suite);
+
+			count++;
+			ptr->metadata.run_count++;
+			ptr->metadata.fail_count += fail ? 1 : 0;
```

Comment:
Here we modify `metadata` which is a member of `const`-qualified/rodata(ROM)-allocated data. We cannot do this.